### PR TITLE
Leaves - Mariya

### PR DESCRIPTION
**Pull request by M-Burr**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model deals with the business logic of the program. It also reads & modifies the database & allows us to do so using Ruby code.
Describe in your own words what the Controller is doing in Rails | The controller manages interactions between the model and the viewer.
Describe in your own words what the View is doing in Rails | The View provides the visuals that the user can actually see and interact with. 
Describe an edge-case controller test you wrote | I wrote an edge case for the controller actions having invalid ids.
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params add enhanced security & prevent outside users from modifying your forms (i.e., the keys in the param hash). Strong params should be stored in private methods.
How are Rails migrations related to Rails models? | Migrations allow you to create & modify 
Describe one area of Rails that are still unclear on |  I have some confusion about routes.

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    elsif @task.update(name: params[:task][:name], description: params[:task][:description], completion_date: params[:task][:completion_date])
+      redirect_to tasks_path
+      return 
+    else 
+      render :edit
+      return
+    end
+  end 
+  
+  def completed
```
Comment (beccaelenzil) [inline app/controllers/tasks_controller.rb:54]: Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent.

**Comment by beccaelenzil** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | check
Answered comprehension questions | check
Successfully handles: Index, Show | check
Index & Show tests pass | check
Successfully handles: New, Create | check
New & Create tests pass | I wasn't able to run tests, but they look good.
Successfully handles: Edit, Update |check
Tests for Edit & Update test valid & invalid task ids | check
Successfully handles: Destroy, Task Complete | The ways to mark a task complete is a little funny. What does the check mark do? When you mark a task complete, you need to refresh in order for the view to change to indicate the task is complete with a line through it. 
Tests for Destroy & Task Complete include tests for valid and invalid task ids | check
Routes follow RESTful conventions | check
Uses named routes (like `_path`) | check
**Overall** | Good job overall. You hit the learning goals of using partials, using strong params, and following Rails conventions. There's one issue -- the completed action doesn't work quite as expected -- you need to refresh to see the strikethrough and the function of the checkbox is not clear.
